Review

GitSparTV opened the pull request.

RyabovichAM committed:
Add main_frame class

AlikNT committed:
Added irc_er_diagram

AlikNT committed:
Postgres initial commit

AlikNT committed:
Add doc/README_irc_chat_db

RyabovichAM committed:
Add transfer classes

RyabovichAM committed:
Add first widgets iteration to MainFrame

Commit:
Add server classes (Alexei)

Commit:
Add README_IRC - project description

RyabovichAM committed:
Add RoomsFrame gui class

Commit:
Add message send on Enter key press

Commit:
Add domain/src and domain/include folders with shared files for server and client: tokenizer, user_foo, service, const

RyabovichAM committed:
Add general sources to domain

RyabovichAM committed:
Update Cmakelists.txt

RyabovichAM committed:
Update CMakeLists.txt

RyabovichAM committed:
Update tcpclient class: added template version of methods

Commit:
Update server classes (Alexei)

Commit:
Update server classes (Alexei)

Commit:
Update server classes (server, Alexei)

Commit:
Update domain (server, Alexei)

AlikNT committed:
Added tests and some improvements

AlikNT committed:
Update README_irc_chat_db.md

RyabovichAM committed:
Add message_handler class

RyabovichAM committed:
Add pausable_thread class

Commit:
Update domain, Add include/guardlock.h, src/guardlock.cpp (server, Alexei)

Commit:
Update domain and server (HTTP), Add ACTION for all responses, Add server config data/loaderserv.conf

RyabovichAM committed:
Add login gui class

Commit:
Update domain and server/include, server/src (HTTP), updated CMakeLists, Add client example

Commit:
Update domain/include/service.h, domain/src/service.cpp (server, Alexei)

Commit:
Update server/include, server/src (HTTP), Update client example, Add logger and logging functions include/logger.h, include/log_functions.h

AlikNT committed:
Update README_irc_chat_db.md

AlikNT committed:
Update README_irc_chat_db.md

AlikNT committed:
Added pagination and updated README_irc_chat_db.md

RyabovichAM committed:
Add settings frame class

Commit:
Update server/include, server/src (HTTP), Update client example, Update doc/Readme - instruction

RyabovichAM committed:
Update MainFrame connection

Commit:
Update doc/Readme - domain/const description

Commit:
Add password_hasher with OpenSSL to domain/, Update CMakeLists and Readme - OpenSSL instruction

Commit:
Add password_hasher with OpenSSL to domain/, Update CMakeLists and Readme - OpenSSL instruction

AlikNT committed:
Restore CMakeLists.txt from develop

AlikNT committed:
Added IRCDBWrapper class and /doc/README_db_wrapper.md

AlikNT committed:
Restore CMakeLists.txt from develop

RyabovichAM committed:
Add room create logic

RyabovichAM committed:
Add rooms users frame class and logic

RyabovichAM committed:
Add login/SighUp

Commit:
Update CMakeLists OpenSSL for domain

Commit:
Initial commit: Add server files to new branch server-drogon

Commit:
Add logger plugin, Refactor http controllers, Add const, Update doc/README response examples

RyabovichAM committed:
New webSocket vision

RyabovichAM committed:
Add roomlists to roomFrame class

Commit:
Refactor http controllers, Update doc/README with ALL response examples

Commit:
Add server-business-logic: chat_service and chat_models, Update and Refactor all *_controller files

RyabovichAM committed:
Add roomsframe, mainframe endpoints requests

AlikNT committed:
Add required methods to DB and tests for them

RyabovichAM committed:
Refactoring

Commit:
Update server/include, server/src (HTTP), Update client example, Update doc/Readme - instruction

RyabovichAM committed:
Merge branch 'develop' into server-drogon

RyabovichAM committed:
Server drogon (#6)

### Внес правки и скорректировал ответы на запросы:

**JOIN** Сделал {"info" : room_name}.
Вопрос: не лучше ли поле заменить:  {"room" : room_name} ?

**LOGIN** Сделал
{"token":"b9b33f94f897aa77af29cca08d08ddca","user":"ann"}
Вопрос: не очень-то безопасно выходит одновременно отправлять имя и
токен.

RyabovichAM committed:
Conflict resolve

RyabovichAM committed:
Merge with develop

RyabovichAM committed:
Fix save settings on Save button

AlikNT committed:
Added example to README_db_wrapper.md how to connect to real db on virtual machine

RyabovichAM committed:
Fix login_frame class:login info worked incorrect

RyabovichAM committed:
Fix websocket class: incorrect saving of url

RyabovichAM committed:
Fix main_frame class: bug on update text button on gtk

AlikNT committed:
IRC chat DB (#7)

Реализована БД PostgreSQL с методами, покрывающими основные потребности
сервера. Подробное описание можно посмотреть
[README_db_wrapper.md](https://github.com/ya-masterskaya-cpp/q2_2025_chat_project_team_1/blob/irc-db/doc/README_db_wrapper.md),
[README_irc_chat_db.md](https://github.com/ya-masterskaya-cpp/q2_2025_chat_project_team_1/blob/irc-db/doc/README_irc_chat_db.md)

RyabovichAM committed:
Fix room_fream: token transfer error when receiving roomsЭ

RyabovichAM committed:
Client (#8)

Реализовано базовое окно, окно комнат/пользователей, окно настроек. 
 Работает совместно с сервером регистрация, логин, получение комнат.  
В окне настроек сохраняется ip/порт.

AlikNT committed:
Added method GetRoomsMembersByName

Commit:
Add token auth for client requests, Update JOIN room handling, ReadMe_IRC description, CMakeLists with PostgeSQL

RyabovichAM committed:
Added method GetRoomsMembersByName (#9)

Добавил метод GetRoomsMembersByName, тест к нему. Также обновил
описание.

RyabovichAM committed:
Add token auth for client requests, Update JOIN room handling, ReadMe… (#10)

**Изменения по клиенту:**
1. Добавлена авторизация по токену ко всем запросам клиента
(client/src/message_handler.cpp), которые требует сервер.
2. Исправлен метод эндпоинта ROOM_JOIN = "/api/room/join" на сервере:
успех, если пользователь уже в комнате (server/src/room_manager.cpp)
2.1 также обновлёно описание ответа в ReadMe_IRC (doc/ReadMe_IRC.md)

3. Скорректирован CMakeLists.txt для PostgreSQL

RyabovichAM committed:
Fix main_frame: empty message to server

RyabovichAM committed:
Fix main_frame: empty message to server (#12)

RyabovichAM committed:
Fix message_handler: createroom checked 200 instead of 201

Commit:
Fix login and logout server requests, Update ReadMe

RyabovichAM committed:
Fix login and logout server requests, Update ReadMe (#14)

**Изменения по серверу:**

1. Скорректирована обработка входа AuthController::LoginUser:
- вход невозможен при отсутствии регистрации или если пользователь уже
залогинен,

2. Скорректирована обработка выхода AuthController::LogutUser:
- при выходе сбрасывается комната (room_ = nullptr), указанная для
пользователя User

3. Обновлено описание ответа LOGIN в ReadMe_IRC (doc/ReadMe_IRC.md)

RyabovichAM committed:
Fix message_handler: createroom checked 200 instead of 201 (#13)

Commit:
Add PostgreSQL in server IRC (branch irc-db)

RyabovichAM committed:
Add PostgreSQL in server IRC (branch irc-db) (#15)

**Изменения по серверу:**

1. Добавлено логирование всех запросов кроме RoomController::CurrentRoom
и MessageController::SendMessage
2. Добавлено логирование всех событий WebSocket
3. Полностью переработан интерфейс ChatService для вызова методов БД
PostgreSQL
4. Добавлено хранение комнат, пользователей (имя, хэш пароля) в БД
PostgreSQL
5. За хранение токенов авторизации отвечает TokenManager. При
перезапуске сервера потребуется новый вход с тем же паролем и логином
6. Добавлен скрипт для создания таблиц БД, конфиг в data/config.json
Скрипт для создания таблиц
```bash
 ./setup_chat_db.sh
```
Команда для удаления таблиц
```bash
sudo -u postgres dropdb chat_db
```
7. Обновлен ReadMe - метод leave - выбрасывает в комнату general (аналог
join, но всегда в general)

RyabovichAM committed:
Fix mainframe class

RyabovichAM committed:
Fix mainframe class (#16)

1. Установлен минимальный размер для main_frame
2. Реализован disconect при закрытии приложения

RyabovichAM committed:
Update main_frame: change coonect/disconnect buttons

RyabovichAM committed:
Upgrade main_frame class: change panels layouts

RyabovichAM committed:
Update main_frame class: added control buttons functionalities

RyabovichAM committed:
Update main_frame class: currected received messages

litiyltk committed:
Feature/change gui layouts (#17)

1. Поменял лейаут функционала, все на 1м окне теперь

RyabovichAM committed:
Update main_frame class: update info panel, set lbl txt

RyabovichAM committed:
Update main_frame class: update info panel, set lbl txt

litiyltk committed:
Update main_frame class: update info panel, set lbl txt (#18)

Commit:
Add Upload message and Get recent messages feature, Update ReadMe

Commit:
Add Upload message and Get recent messages feature, Update ReadMe

AlikNT committed:
Added docker files

RyabovichAM committed:
Add Upload message and Get recent messages feature, Update ReadMe (#19)

**Обновление по серверу:**

1. Добавлен запрос PING и логирование, PingController::Ping 
2. Добавлен запрос на сохранение отправляемого клиентом сообщения
MessageController::UploadMessage и логирование. Этот запрос выполняется
независимо от отправки MessageController::SendMessage
3. Добавлен запрос на выгрузку последних N сообщений
MessageController::GetRecentMessages и логирование. Этот запрос нужен
клиенту для получения истории сообщений в комнате (далее, можно добавить
постраничный вывод, см. IRCDBWrapper::GetRoomMessagesPage)
4. Добавлено логирование оставшихся запросов RoomController::CurrentRoom
и MessageController::SendMessage
5. Обновлен ReadMe описанием новых запросов
6. Временно добавлен костыль http_utils::TrimNewLineSymb для среза '\n'
для имени комнаты в RoomController::JoinRoom и
RoomController::ListUsersInRoom

RyabovichAM committed:
Fix main_frame: get rooms error, get users in room error requests

RyabovichAM committed:
Fix main_frame: get rooms error, get users in room error requests

RyabovichAM committed:
Update message_handler, main_frame: implemented getting the latest room messages

RyabovichAM committed:
Update message_handler, main_frame: implemented getting the latest room messages

RyabovichAM committed:
Fix main_frame: get rooms error, get users in room error requests (#20)

AlikNT committed:
Docker is working correctly

AlikNT committed:
Adde README_DOCKER.md

RyabovichAM committed:
Irc db (#21)

1. Добавил докер файлы.
2. Изменил CMakeLists.txt. Добавил тесты БД, изменил поведение с
клиентом так, чтобы в докере он не собирался. Может быть конфликт,
поэтому нужно принять мои изменения.
3. Изменил способ подключения к БД в коде сервера. Сделал через
переменные подключения, передаваемые в докер через файл .env. Этот файл
я не коммитил, т.к. в нем пароли. Брать строку подключения и
конфигурационного файла не безопасно, поэтому сделал так. Старые строки
не удалил, а закомментировал. Сам .env файл положу на виртуалку руками.
Поэтому могут быть конфликты, нужно принять мою версию.
4. В докере изменился путь к конфигурационному файлу /data/config.json,
поэтому пришлось изменить путь в main.cpp. Старую строку
закомментировал, нужно принять мои изменения.
5. Добавил файл описания README_DOCKER.md
6. Ну меня на Ubuntu 24.04 докер запустился, тесты БД отработали
корректно.

RyabovichAM committed:
Update: client refactoring

RyabovichAM committed:
Merge branch 'develop' into client

RyabovichAM committed:
Update: client refactoring (#22)

AlikNT committed:
Changed CMakeLists.txt

Removed REQUIRED from line "find_package(vxWidgets...)"

AlikNT committed:
Update config.json

Удалил параметр для подключению к БД

AlikNT committed:
Update docker-compose.yml

Добавил проброс порта для сервера

AlikNT committed:
Changed listener address from 127.0.0.1 to 0.0.0.0

RyabovichAM committed:
Update main_frame : constructor refactoring

RyabovichAM committed:
Merge branch 'develop' into client

Commit:
Add docs to server, Alexei

AlikNT committed:
Fixed some bugs

AlikNT committed:
Fixed some bugs

AlikNT committed:
Fixed config.json

AlikNT committed:
Add docs to server, Alexei (#23)

**Обновление по серверу:**

1. Добавлено описание по серверу, Алексей
2. Скорректирована ошибка при падении клиента: если закрыто соединение,
то принудительный логаут в ChatWebSocket::handleConnectionClosed
3. Удалён костыль http_utils::TrimNewLineSymb для среза '\n' для имени
комнаты в RoomController::JoinRoom и RoomController::ListUsersInRoom
4. CMakeLists.txt добавлен wxWidgets COMPONENTS core base richtext xml
html при сборке на Ubuntu 20.04
5. Удалена закомментированная строка с неактуальным путём к конфигу в
main.cpp
6. Удален неиспользуемый конфиг для клиента со списком серверов
servers.json

AlikNT committed:
Fixed som bugs

AlikNT committed:
Fixed some bugs

Commit:
Fix ReadMe and controllers REST API, Add uml-diagram for server

AlikNT committed:
Fix ReadMe and controllers REST API, Add uml-diagram for server (#29)

**Исправления после тестирования REST API**

1 - POST /api/v1/auth/login - исправил порядок на token, user (описание
doc/ReadMe_IRC.md)
2 - POST /api/v1/room/join - исправил код 400 на 404 NotFound для
случая, когда запрашиваемая комната не найдена (описание
doc/ReadMe_IRC.md и логика server/src/room_controller.cpp)
3 - POST api/v1/room/leave - исправил код 404 для случая, когда комната
general не найдена (описание doc/ReadMe_IRC.md и логика
server/src/room_controller.cpp)
Важно! Если пользователь делает join из одной и той же комнаты -
результат 200. Аналогично для leave из general.
4 - POST /api/v1/messages/send - только внутрикомнатная рассылка,
адресата нет, удалил поле "to" (описание doc/ReadMe_IRC.md)
5 - POST /api/v1/messages/upload - корректно, код 500 при ошибки
сохранения в БД, внутренняя ошибка БД.
6, 7, 8 -
/api/v1/messages/recent?room=<ИМЯ_КОМНАТЫ>&max_items=<НАТУРАЛЬНОЕ_ЧИСЛО>
- переработал логику проверки, описание в doc/ReadMe_IRC.md

**Документация**

Добавлена uml-диаграмма по серверу (doc/irc_server_diagram)

Commit:
FIX websocket json message parsing logic, Update ReadMe websockets descriptions

RyabovichAM committed:
FIX websocket json message parsing logic, Update ReadMe websockets de… (#30)

**Исправления по серверу**

- исправлена ошибка парсинга json-сообщения от клиента WebSocket,
приводящая к падению сервера

**Обновления описания**

- добавлен блок описания логики логина-логаута и WebSocket

Commit:
FIX cyrillic messages by client

RyabovichAM committed:
FIX cyrillic messages by client (#31)

**Исправления по клиенту**

- исправлена обработка кириллицы
- исправлена ошибка при парсинге JSON
- добавлено закрытие окна с конфигом сервера
- добавлен логаут после нажатия кнопки дисконекта
- изменена проверка Xlib под Linux

RyabovichAM committed:
Refactoring Cmakelists: added subdirectories projects

RyabovichAM committed:
Merge branch 'develop' into client

litiyltk committed:
Refactoring Cmakelists.txt (#32)

1.Разбил CMakelist.txt на подпроекты, каждый проект сервер, клиент может
собираться самостоятельно
2. mainframe небольшой рефакторинг

Commit:
Add API GetRoomMessagesPage Update ReadMe, Add ChatNotifier, Refactor WS dependencies

RyabovichAM committed:
Refactoring Cmakelists: added subdirectories projects

RyabovichAM committed:
Add API GetRoomMessagesPage Update ReadMe, Add ChatNotifier, Refactor… (#33)

**Обновления по серверу**

1. Добавлен новый запрос REST API, метод контроллера
MessageController::GetRoomMessagesPage для получения страницы сообщений
из указанной комнаты с использованием параметров offset и limit
(смещение относительно начала списка сообщений и максимальное количество
сообщений, возвращаемое в ответе).

2. ReadMe_IRC.md дополнен описанием запроса, примерами возможных
ответов.

3. Введён ChatNotifier для изоляции работы с WebSocket - теперь
REST-контроллеры не используют напрямую WebSocket, а имеют доступ через
ChatNotifier - меньше связей слоёв.

4. По файлам WebSocket'а:
- переименован chat_websocket.* -> chat_ws_controller.*
- chat_notifier.h - новый файл.
- user_controller.cpp - использует ChatNotifier
- msg_controller.cpp - использует ChatNotifier

**Обновления по клиенту**

1. В CMakeLists для клиента добавлены компоненты html xml richtext в
блок find_package

RyabovichAM committed:
Merge branch 'develop' into client

RyabovichAM committed:
Fix login_frame, create_room_frame fixed sizes

RyabovichAM committed:
Fix settings_frame size bug

RyabovichAM committed:
Fix default sizes bug (#35)

1.Классы create_room_frame, login_frame устранена проблема не
отображении кнопок при фиксированном размере.
2. create_room_frame добавлено событие создания комнаты при нажатии
enter
3. login_frame добавлено событие смена фокуса при нажатии enter на поле
ввода имени

litiyltk commented on server/src/chat_service.cpp:
```diff
@@ -0,0 +1,198 @@
+        return false;
+    }
+
+    auto room_opt = GetCurrentRoomName(token);
```
Хорошая мысль, исправлю. Клиент же знает свою комнату

RyabovichAM committed:
Client manifest flatpak

RyabovichAM committed:
Merge branch 'develop' into client

litiyltk commented on server/src/msg_controller.cpp:
```diff
@@ -0,0 +1,154 @@
+
+    int max_items = 0;
+    try {
+        max_items = std::stoi(max_items_str);
```
Скорректировал

RyabovichAM commented on client/CMakeLists.txt:
```diff
@@ -0,0 +1,61 @@
+
+
+    if(NOT DEFINED DOMAIN_SET)
+        message(STATUS "Переменная MY_VARIABLE существует и имеет значение: ${MY_VARIABLE}")
```
клиент и сервер могут собираться самостоятельно, это проверка что библиотека domain не устанавливается из общего cmakelists.txt. Если переменная не установлена, значит domain не собрана, и ее нужно собрать для конкретоной цели.

GitSparTV replied:
Они могут собираться и по отдельности, но корневой CMakeLists будет ведь тем же.

Просто указываешь цель client и он его билдит, по умолчанию билдит all

RyabovichAM commented on client/include/message_handler.h:
```diff
@@ -0,0 +1,59 @@
+    void SetUrl(const std::string& url);
+
+private:
+    UserData& user_;
```
да, message_handler не владеет данными пользователя

litiyltk commented on server/src/chat_websocket.cpp:
```diff
@@ -0,0 +1,164 @@
+    }
+
+    if (token.empty()) {
+        drogon::app().getPlugin<LoggerPlugin>()->LogWebSocketEvent("Empty token");
```
Логгер надо весь пересмотреть. Логирование действий по вебсокет-соединениям в лоб сделал

litiyltk commented on LICENSE:
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Github сам предложил, когда репу сервера создавал))

RyabovichAM committed:
Update main_frame: add messages autoscroling

litiyltk committed:
Autoscrolling (#44)

Update main_frame: added messages autoscrolling

Commit:
Update Server after review, Refactor ChatNotifier

litiyltk commented on server/include/chat_service.h:
```diff
@@ -0,0 +1,180 @@
+        /// @brief Указатель на объект `IRCDBWrapper` для доступа к базе данных.
+        std::shared_ptr<IRCDBWrapper> db_wrapper_;
+        /// @brief Указатель на объект `TokenManager` для управления токенами.
+        std::shared_ptr<TokenManager> token_manager_; // Токены сбрасываются при падении сервера, требуется новый вход по сохраненным в БД логину и хэшу пароля
```
Агрегации достаточно, не создаются в конструкторе класса

litiyltk commented on server/include/http_utils.h:
```diff
@@ -0,0 +1,139 @@
+                               const std::function<void(const drogon::HttpResponsePtr&)>& callback) {
+    Json::Value body;
+    body["info"] = message;
+    drogon::app().getPlugin<LoggerPlugin>()->LogResponse(message);
```
Отметил

litiyltk commented on server/src/ping_controller.cpp:
```diff
@@ -0,0 +1,6 @@
+
+
+void PingController::Ping(const drogon::HttpRequestPtr &, std::function<void(const drogon::HttpResponsePtr &)> &&callback) {
+    http_utils::RespondWithSuccess("Ping success", drogon::k200OK, std::move(callback));
```
Отметил

AlikNT commented on server/src/postgres.cpp:
```diff
@@ -0,0 +1,395 @@
+}
+
+std::optional<ConnectionPool::ConnectionWrapper> ConnectionPool::GetConnection(std::chrono::milliseconds timeout) {
+    std::unique_lock lock{mutex_};
```
Поскольку это количество одновременных подключений, для того, чтобы избежать перегрузки сервера бд.

RyabovichAM committed:
Fix login_frame: utf8 input was shown as empty

RyabovichAM committed:
Fix create_room_frame: utf8 input was shown as empty

litiyltk committed:
login_frame, create_room_frame: fix utf8 input (#46)

RyabovichAM committed:
Update: webSocket_class was refactred

RyabovichAM committed:
Update Server after review, Refactor ChatNotifier (#45)

**Обновления сервера**
- корректировки после ревью
- рефакторинг ChatNotifier

RyabovichAM committed:
Update client_tests cmakelist

Commit:
Update Server with token ttl, Update logger with log-levels, log-files, Refactor ChatNotifier

Commit:
Docker: disable client build, fix Server/DbTests paths

litiyltk committed:
Update Server with token ttl, Update logger with log-levels, log-file… (#47)

**Обновления по серверу**

1. Добавлена поддержка времени жизни токена и интервал проверки:
обновлен TokenManager, есть удаление токена, удаление из комнат и
дисконект WS по таймеру. Параметры задаются в конфиге.
TokenMonitorPlugin отвечает за проверку валидности по таймеру.
2. Переработан LoggerPlugin: добавлена поддержка уровней логирования в
конфиге, корректный вывод в файл по заданным параметрам в конфиге.
3. Отрефакторен ChatNotifier для изоляции работы с WebSocket.

**Обновления Docker**

4. Обновлен Dockerfile и docker-compose.yml: сборка GUI-клиента
исключена из образа контейнера через флаг BUILD_CLIENT=OFF, бинарники
Server и DbTests теперь собираются в подкаталог build/server/.
5. Добавлен флаг BUILD_CLIENT в CMakeLists.txt для опциональной сборки
клиента вне Docker.

RyabovichAM commented on client/src/main_frame.cpp:
```diff
@@ -0,0 +1,429 @@
+        ws_client_->Run();
+        is_connected_ = true;
+    } else {
+        chat_history_->AppendText("You are connected already.\n");
```
как сделать if guard, макросом?

GitSparTV replied:
if guard называют вот это:

```cpp
if (чего-то нет) {
    // Дальше код только для случаев с когда всё есть
    return {};
}

// Код для основного случая
// ...

return ...;
```

Сейчас же здесь вот так:

```cpp
if (всё есть) {
    // Код для основного случая
    // ...

    return ...;
} else {
    return {};
}
```

if guard позволяет убрать вложенность

RyabovichAM committed:
Refactoring flatpak mainfest

Commit:
Refactor ChatService to return messages in reverse order

RyabovichAM committed:
Refactor ChatService to return messages in reverse order (#51)

**Обновления по серверу**

Методы ChatService::GetRecentMessages и ChatService::GetRoomMessagesPage
возвращают список сообщений в порядке от новых к старым

RyabovichAM committed:
Merge branch 'develop' into client

RyabovichAM committed:
Refactor: add query_handler class, change message handler

RyabovichAM committed:
Chore: add tests to message handler class

RyabovichAM committed:
Client refactoring (#52)

1. Добавлены тесты для web_socket, message_handler
2. добавлен класс обработки/сохранения запросов.
3. мелкие исправления

RyabovichAM committed:
feat: main_frame clear chat history when join a room

RyabovichAM committed:
refactor: main_frame change if_guard is_connected

RyabovichAM committed:
feat: Add show rooms at main frame when authorized success

RyabovichAM committed:
fix: main frame join room after update rooms list

RyabovichAM committed:
Client fix, refactor, feature (#53)

Commit:
Refactor PasswordHasher - add std::array, Refactor http_utils.h and chat_notifier.h - split .h/.cpp

RyabovichAM committed:
Refactor PasswordHasher - add std::array, Refactor http_utils.h and c… (#54)

**Обновления domain**

Заменён unsigned char hash[SHA256_DIGEST_LENGTH] на std::array<unsigned
char, SHA256_DIGEST_LENGTH> в классе PasswordHasher

**Обновления по серверу**

- http_utils.h разделён на заголовочный файл http_utils.h и файл
реализации http_utils.cpp
- chat_notifier.h разделён на заголовочный файл chat_notifier.h и файл
реализации chat_notifier.cpp
